iothub/device/src/Transport/AmqpIoT/AmqpIoTErrorAdapter.cs

```diff
@@ -239,7 +239,7 @@ public static Exception ToIotHubClientContract(Error error)
             }
             else if (error.Condition.Equals(AmqpErrorCode.ResourceLimitExceeded))
             {
-                retException = new DeviceMaximumQueueDepthExceededException(message);
+                retException = new QuotaExceededException(message);
```

Comment:
We need to fully vet this breaking change, run it by Mark, and bring alternatives.

Reply:
Yes. Starting the conversation now. Will hold on to this PR until we have a resolution.

Reply:
Under what scenarios does the service throw `amqp:resource-limit-exceeded` on amqp? 
You can update the doc [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/devdoc/amqpTransportExceptions.md#not-retried-by-the-sdk), so that it carries an explanation for `amqp:resource-limit-exceeded` (right now it says - "Inspect the exception details, collect logs and contact service"; but if we have more info, it would help to be more descriptive.)